src/tax/basis.test.js

```diff
@@ -0,0 +1,67 @@
+// @flow
+
+import moment from "moment";
+import type {BasisFrame} from "./basis";
+import {LIFOCostBasisCalculator} from "./basis";
+
+describe("cost basis calculation", () => {
+  const dayFromYear = (x) =>
+    moment().set({
+      year: x,
+      month: 0,
+      day: 0,
+      hour: 0,
+      minute: 0,
+      second: 0,
+      millisecond: 0,
+    });
+  const date0 = dayFromYear(1990);
```

Comment:
Moments are mutable. These declarations introduce mutable state shared among tests, under the tacit assumption that it will never be modified. This is a bad idea. If you want to keep these dates, consider instead `const date0 = () => dayFromYear(1990)` and so on. However, just inlining them seems to me like it might be clearer, anyway.

As for the mutability, you might like to write some tests indicating that the calculator does not mutate its input.